### PR TITLE
Avoid zipping empty directory

### PR DESCRIPTION
**Pull request by blindpirate**
When there're file handle leaking and some empty directories happen to exist in the tmp test directory:

```
+-- LeakingTest
  +--- test directory
    +--- leaked files
  +--- empty directory 1
  +--- empty directory 2
```

The empty directories are zipped into published artifact, which makes it difficult to find out the real leaking directory:

image

This PR deletes the empty directories before zipping them into the published artifact.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test QF

File: buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts [hunk 1]
```diff
@@ -151,9 +151,17 @@ fun Task.attachedReportLocations() = when (this) {
     else -> emptyList()
 }
 
+fun File.isEmptyDirectory() = list()?.isEmpty() == true
+
 fun prepareReportForCiPublishing(report: File, projectName: String) {
     if (report.exists()) {
         if (report.isDirectory) {
+            report.listFiles()?.forEach {
+                if (it.isEmptyDirectory()) {
```
[the remark this answers is not in the document]
Comment (blindpirate) [reply buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts:160]: Unfortunately I don't. I spent some time on this, it works pretty well, the whole test directory is deleted - I don't understand why there're leftover empty directories so I decide to workaround this issue on another end.

[the remark this answers is not in the document]
Comment (wolfs) [reply buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts:160]: Is the problem maybe related to multiple tests failing? Here, there are 38 test failures, and only one zip: https://builds.gradle.org/buildConfiguration/Gradle_Check_WatchFs_30_bucket46/36733372?buildTab=overview

Maybe we clean up the directories without zipping them?

[the remark this answers is not in the document]
Comment (blindpirate) [reply buildSrc/subprojects/buildquality/src/main/kotlin/gradlebuild/ci-reporting.gradle.kts:161]: Note that the exception is `java.nio.file.AccessDeniedException`, another possibility is that these directories are not cleaned up, but somehow the zipping process can't access their contents so they look like empty. 

Anyway, let's add more log to it to get more details.

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by wolfs** [comment]
@bot-gradle test QF

**Comment by wolfs** [comment]
@blindpirate I pushed some changes so we are only packaging up the files and the packaging works even if some strange symlinks exist in the directory. Please take a look!

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
@wolfs LGTM, but I can't approve myself's PR. I also did a minor refactoring, if it's good let's merge it.

**Pull request merged**